### PR TITLE
[DOCS] MistralAI function calling docs

### PR DESCRIPTION
**Pull request by czelabueno**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new model selection feature in MistralAI with updated model classifications.
	- Added functionality for connecting to external tools via function calling in MistralAI.
	- New fields and methods added to support tool execution requests and responses in MistralAI's chat and streaming chat models.
	- Introduced new data models for handling tool and function calls within MistralAI.
	- Updated MistralAI documentation to reflect new features and deprecations.

- **Enhancements**
	- Enhanced handling of different message types and tool calls in MistralAI.
	- Improved response format handling for MistralAI chat completions.

- **Refactor**
	- Refactored various classes to handle new model names, tool specifications, and response formats.

- **Documentation**
	- Updated MistralAI integration documentation with new model selection features and deprecations.

- **Tests**
	- Enhanced tests to cover new tool execution scenarios and JSON object response formats.
	- Updated model names and configurations in tests.

### CODE REVIEW

File: docs/docs/integrations/language-models/mistralai.md [hunk 1]
```diff
@@ -20,9 +20,9 @@ For Maven project `pom.xml`
 </dependency>
 
 <dependency>
-    <groupId>dev.langchain4j</groupId>
-    <artifactId>langchain4j-mistral-ai</artifactId>
-    <version>{your-version}</version>
+<groupId>dev.langchain4j</groupId>
```
Comment (langchain4j) [inline docs/docs/integrations/language-models/mistralai.md]:
```suggestion
    <groupId>dev.langchain4j</groupId>
```

File: docs/docs/integrations/language-models/mistralai.md [hunk 2]
```diff
@@ -20,9 +20,9 @@ For Maven project `pom.xml`
...
+<artifactId>langchain4j-mistral-ai</artifactId>
```
Comment (langchain4j) [inline docs/docs/integrations/language-models/mistralai.md]:
```suggestion
    <artifactId>langchain4j-mistral-ai</artifactId>
```

File: docs/docs/integrations/language-models/mistralai.md [hunk 3]
```diff
@@ -20,9 +20,9 @@ For Maven project `pom.xml`
...
+<version>{your-version}</version>
```
Comment (langchain4j) [inline docs/docs/integrations/language-models/mistralai.md]:
```suggestion
    <version>{your-version}</version>
```

**Review by ghost — DISMISSED** [review]
@czelabueno great job, thank you!

**Pull request merged**